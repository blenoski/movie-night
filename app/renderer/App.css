body {
  background-color: #141414;
  min-width: 320px;
}

.App {
  background-color: #141414;
  width: 100%;
}

.App-header {
  background-color: rgb(20,20,20);
  background-color: rgba(20,20,20,0.7);
  color: white;
  padding: 10px 10px 10px 20px;
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  font-family: CopperPlate, Times;
  font-size: 1.5em;
  display: flex;
  justify-content: space-between;
  flex-wrap: nowrap;
}

h2 {
  line-height: 1.3;
  margin: 0;
  display: block;
  font-size: 1.2em;
  font-weight: bold;
  color: #999;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  background-color: transparent;
  text-transform: capitalize;
  cursor: pointer;
}

h2:hover {
  color: #FFF;
}

.movie:active {
  border: 5px solid rgba(255,255,255,0.5);
}

.play-button {
  background: rgba(0,0,0,0.1);
  border: 3px solid rgba(255,255,255,0.9);
  border-radius: 50%;
  height: 80px;
  width: 80px;
  padding-left: 5%;
  color: rgba(255,255,255,0.9);
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.play-button:hover {
  color: rgb(2,117,216);
  background-color: rgba(0,0,0,0.5);
}

.mscroll::-webkit-scrollbar {
    height: 10px;
}

.mscroll::-webkit-scrollbar-button {
  display: none;
}

.mscroll::-webkit-scrollbar-track {
    display: none;
}

.mscroll::-webkit-scrollbar-track:hover {
  display: block;
  border-radius: 10px;
  box-shadow: inset 0 0 4px rgba(255,255,255,0.4);
}

.mscroll::-webkit-scrollbar-thumb {
    display: block;
    border-radius: 10px;
    box-shadow: inset 0 0 4px rgba(255,255,255,0.4);
}
