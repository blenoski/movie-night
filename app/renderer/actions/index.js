import { SEARCH_DIRECTORY } from './types'

export function updateSearchDirectory (directory) {
  return {
    type: SEARCH_DIRECTORY,
    payload: directory
  }
}
