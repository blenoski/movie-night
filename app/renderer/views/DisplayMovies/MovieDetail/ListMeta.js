import React from 'react'
import styled from 'styled-components'

export default ({ actors, director, genres }) =>
  <Section>
    <Item>
      <Heading>Starring:</Heading> {actors.join(', ')}
    </Item>
    <Item>
      <Heading>Director:</Heading> {director}
    </Item>
    <Item>
      <Heading>Genres:</Heading> {genres.join(', ')}
    </Item>
  </Section>

const Section = styled.section`
  margin-top: 24px;
`

const Heading = styled.span`
  color: rgba(255, 255, 255, 0.9);
`

const Item = styled.h6`
  text-transform: capitalize;
`
