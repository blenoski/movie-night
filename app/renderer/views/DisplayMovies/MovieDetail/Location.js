import React from 'react'
import styled from 'styled-components'
import { FileIcon } from '../../../icons'

export default ({ location, handleClick }) =>
  <Section onClick={handleClick}>
    <StyledFileIcon video large />
    <h6>{location}</h6>
  </Section>

const Section = styled.section`
  display: flex;
  cursor: pointer;
  margin-top: 30px;
  &:hover {
    color: rgba(255,255,255,0.9);
  }
`

const StyledFileIcon = styled(FileIcon)`
  margin-right: 5px;
  `
