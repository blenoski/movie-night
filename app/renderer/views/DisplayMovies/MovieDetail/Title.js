import React from 'react'
import styled from 'styled-components'

export default ({ title }) =>
  <Title>{title}</Title>

const Title = styled.h1`
  color: rgba(255,255,255,1);
`
