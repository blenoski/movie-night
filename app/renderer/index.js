import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import App from './App'
import store from './AppState'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
