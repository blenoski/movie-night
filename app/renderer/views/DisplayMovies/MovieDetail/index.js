import fs from 'fs'
import React, { Component } from 'react'
import styled from 'styled-components'
import { shell } from 'electron'
import { fileExists, filePathToUrl } from '../../../../shared/utils'
import ListMeta from './ListMeta'
import Location from './Location'
import Meta from './Meta'
import PlayMovieButton from './PlayMovieButton'
import Ratings from './Ratings'
import Title from './Title'
import { Close } from '../../../icons'

export default class MovieDetail extends Component {
  constructor (props) {
    super(props)
    this.openMovieInDefaultPlayer = this.openMovieInDefaultPlayer.bind(this)
    this.showMovieInFinder = this.showMovieInFinder.bind(this)
    this.updateFileAvailable = this.updateFileAvailable.bind(this)
    this.close = this.close.bind(this)
    this.anchor = null
    this.state = { fileAvailable: true }
  }

  componentDidMount () {
    this.props.center &&
    this.anchor &&
    this.anchor.scrollIntoViewIfNeeded() // centers anchor in viewport

    this.updateFileAvailable()
    fs.watchFile(this.props.movie.fileInfo[0].location, this.updateFileAvailable)
  }

  componentWillUnmount () {
    fs.unwatchFile(this.props.movie.fileInfo[0].location, this.updateFileAvailable)
  }

  updateFileAvailable () {
    fileExists(this.props.movie.fileInfo[0].location)
      .then(result => this.setState({ fileAvailable: result }))
  }

  openMovieInDefaultPlayer (e) {
    e.preventDefault()
    const { movie } = this.props
    shell.openItem(movie.fileInfo[0].location)
  }

  showMovieInFinder (e) {
    e.preventDefault()
    const { movie } = this.props

    shell.showItemInFolder(movie.fileInfo[0].location)
  }

  close (e) {
    e.preventDefault()
    const { movie, handleCloseMovieDetails } = this.props
    if (handleCloseMovieDetails) {
      handleCloseMovieDetails(movie)
    }
  }

  render () {
    const { movie } = this.props

    return (
      <FlexboxDiv >
        <button /* Hook for scrolling top of div to center of viewport */
          ref={(node) => { if (node) this.anchor = node }}
          style={{visibility: 'hidden'}}
        />

        <Poster fileUrl={filePathToUrl(movie.imgFile)}>
          {this.state.fileAvailable && <PlayMovieButton onClick={this.openMovieInDefaultPlayer} />}
        </Poster>

        <MovieDetailsContainer>
          <header>
            <Div>
              <Title title={movie.title} />
              <CloseButton onClick={this.close} />
            </Div>
            <Meta year={movie.year} rated={movie.rated} runtime={movie.runtime} />
            <Ratings
              audience={Number(movie.imdbRating) / 10.0}
              critics={Number(movie.metascore) / 100.0}
            />
          </header>

          <VerticalScrollSection>
            {movie.plot}
          </VerticalScrollSection>

          <ListMeta
            actors={movie.actors}
            director={movie.director}
            genres={movie.genres}
          />

          <Location
            location={movie.fileInfo[0].location}
            handleClick={this.showMovieInFinder}
            fileExists={this.state.fileAvailable}
          />
        </MovieDetailsContainer>

      </FlexboxDiv>
    )
  }
}

const FlexboxDiv = styled.div`
  display: flex;
  margin-top: 30px;
`
const Div = styled.div`
  display: flex;
  justify-content: space-between;
`

const CloseButton = styled(Close)`
  color: rgba(255, 255, 255, 0.9);
  cursor: pointer;
  font-size: 30px;
  padding: 5px 0 0;
`

const Poster = styled.aside`
  align-items: center;
  background: ${props => `url(${props.fileUrl}) no-repeat center`};
  display: flex;
  flex-shrink: 0;
  height: 444px;
  justify-content: center;
  margin-right: 20px;
  width: 300px;
`

const MovieDetailsContainer = styled.article`
  color: rgba(255, 255, 255, 0.7);
  display: flex;
  flex-direction: column;
  height: 444px;
  min-width: 500px;
  max-width: 600px;
`

const VerticalScrollSection = styled.section`
  margin-top: 30px;
  overflow-y: auto;
`
