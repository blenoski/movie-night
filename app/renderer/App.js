import React, { Component } from 'react'
import styled from 'styled-components'
import AppController from './AppController'
import DisplayMovies from './display-movies'
import ImportMovies from './import-movies'
import Logo from './logo'
import SearchMovies from './search-movies'

export default class App extends Component {
  render () {
    return (
      <Application>
        <Header>
          <Logo />
          <AppControls>
            <SearchMovies />
            <ImportMovies onClick={AppController.importMovies} />
          </AppControls>
        </Header>
        <DisplayMovies />
      </Application>
    )
  }
}

const Application = styled.div`
  background: rgb(20,20,20);
  padding-top: 40px;
`

const Header = styled.header`
  background-color: rgba(20,20,20,0.7);
  color: white;
  display: flex;
  flex-wrap: nowrap;
  font-family: CopperPlate, Times;
  font-size: 1.5em;
  justify-content: space-between;
  left: 0;
  right: 0;
  top: 0;
  padding: 10px 10px 10px 20px;
  position: fixed;
`

const AppControls = styled.div`
  display: flex;
`
