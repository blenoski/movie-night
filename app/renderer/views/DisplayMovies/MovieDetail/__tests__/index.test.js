'use strict'
/* globals jest, describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import { shellMock } from '../../../../../../__mocks__/electron'
import MovieDetail from '../index'

jest.mock('../../../../../shared/utils', () => {
  return {
    fileExists: () => Promise.resolve(true)
  }
})

const movie = {
  actors: ['Ben Stiller', 'Will Ferrell'],
  director: 'Todd Phillips',
  imgFile: '/path/to/img',
  fileInfo: [
    { location: '/path/to/movie' }
  ],
  genres: ['action', 'comedy', 'drama']
}

describe('Movie Details', () => {
  test('renders', () => {
    const handleClose = jest.fn()
    const component = renderer.create(
      <MovieDetail
        movie={movie}
        handleCloseMovieDetails={handleClose}
        center
      />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Clicking on close button calls back handler
    const close = tree.children[2].children[0].children[0].children[1]
    close.props.onClick({ preventDefault: jest.fn() })
    expect(handleClose).toHaveBeenCalledWith(movie)

    // Clicking on file calls electron.shell openFileInViewer
    const location = tree.children[2].children[3]
    location.props.onClick({ preventDefault: jest.fn() })
    expect(shellMock).toHaveBeenLastCalledWith(movie.fileInfo[0].location)

    shellMock.mockClear()
    expect(shellMock).not.toHaveBeenCalled()

    // Clicking on play button calls electron.shell openItem
    const playButton = tree.children[1].children[0].children[0]
    playButton.props.onClick({ preventDefault: jest.fn() })
    expect(shellMock).toHaveBeenLastCalledWith(movie.fileInfo[0].location)
  })

  test('unmounts', () => {
    const component = renderer.create(
      <MovieDetail
        movie={movie}
      />
    )

    setTimeout(() => component.unmount(), 0)
  })
})
