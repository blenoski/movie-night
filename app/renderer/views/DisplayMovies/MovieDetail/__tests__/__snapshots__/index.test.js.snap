// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details renders 1`] = `
<div
  className="sc-jTzLTM dXWQgN"
>
  <button
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />
  <aside
    className="sc-cSHVUG fYQdoh"
  >
    <div
      className="sc-iwsKbI dBXHBj"
      onClick={undefined}
    >
      <i
        className="fa fa-play sc-gZMcBi cTlzSH"
        onClick={[Function]}
      />
    </div>
  </aside>
  <article
    className="sc-kAzzGY knOsuf"
  >
    <header>
      <div
        className="sc-fjdhpX gddroE"
      >
        <h1
          className="sc-VigVT iqpAPW"
        />
        <i
          className="fa fa-close sc-jzJRlG jpkopr"
          onClick={[Function]}
        />
      </div>
      <h6>
        <span
          className="sc-dnqmqq jzOmvl"
        />
      </h6>
      <h6>
        <span
          className="sc-gqjmRU bZhlDI"
        >
          Audience: 
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
        </span>
        <span
          className="sc-gqjmRU bZhlDI"
        >
          Critics: 
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
          <i
            className="fa fa-star-o"
          />
        </span>
      </h6>
    </header>
    <section
      className="sc-chPdSV eJkNci"
    />
    <section
      className="sc-bdVaJa fyrQKn"
    >
      <h6
        className="sc-htpNat jeDJyY"
      >
        <span
          className="sc-bwzfXH fyPPwG"
        >
          Starring:
        </span>
         
        Ben Stiller, Will Ferrell
      </h6>
      <h6
        className="sc-htpNat jeDJyY"
      >
        <span
          className="sc-bwzfXH fyPPwG"
        >
          Director:
        </span>
         
        Todd Phillips
      </h6>
      <h6
        className="sc-htpNat jeDJyY"
      >
        <span
          className="sc-bwzfXH fyPPwG"
        >
          Genres:
        </span>
         
        action, comedy, drama
      </h6>
    </section>
    <section
      className="sc-ifAKCX iGlUin sc-bxivhb iwqtOS"
      onClick={[Function]}
    >
      <i
        className="fa fa-file-video-o sc-EHOje jhGXeA fa-lg"
      />
      <h6>
        /path/to/movie
      </h6>
    </section>
  </article>
</div>
`;
